generator/restriction_collector.cpp

```diff
@@ -0,0 +1,162 @@
+#include "generator/restriction_collector.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/fstream.hpp"
+
+namespace
+{
+char const kNo[] = "No";
+char const kOnly[] = "Only";
+char const kDelim[] = ", \t\r\n";
+
+bool ParseLineOfNumbers(strings::SimpleTokenizer & iter, vector<uint64_t> & numbers)
+{
+  uint64_t number = 0;
+  for (; iter; ++iter)
+  {
+    if (!strings::to_uint64(*iter, number))
+      return false;
+    numbers.push_back(number);
+  }
+  return true;
+}
+}  // namespace
+
+namespace routing
+{
+RestrictionCollector::RestrictionCollector(string const & restrictionPath,
+                                           string const & featureIdToOsmIdsPath)
+{
+  if (!ParseFeatureId2OsmIdsMapping(featureIdToOsmIdsPath))
```

Comment:
You need MY_SCOPE_GUARD only if you use two individual ifs. Otherwise, you may get rid of MY_SCOPE_GUARD at all, please, re-read my first comment - the first version is short and easy-to-read.